### DISCUSSION TITLE
Use destructuring syntax once we establish Julia v1.7 as the base

### DISCUSSION THREAD

**Discussion by dmbates** [JuliaStats/MixedModels.jl#555] [Ideas]
I often start functions by extracting fields from, e.g. a `LinearMixedModel` struct and Julia v1.7 introduces the destructuring syntax

* `(; a, b) = x` can now be used to destructure properties `a` and `b` of `x`.
  This syntax is equivalent to `a = getproperty(x, :a); b = getproperty(x, :b)` ([#39285]).

**Comment by palday** [comment]
That's some nice syntax. :smile: 

If 1.6 is declared LTS, then we should only do this if `Compat.jl` provides a way to support it.
